skorch/utils.py

```diff
@@ -415,3 +415,22 @@ def store_step(self, step):
     def get_step(self):
         """Return the stored step."""
         return self.step
+
+
+def filter_requires_grad(pgroups):
+    """Returns parameter groups where parameters with
+    ``requires_grad==False`` are filtered out.
+    """
+    for pgroup in pgroups:
+        output = {k: v for k, v in pgroup.items() if k != 'params'}
```

Comment:
Why not use `params = pgroup.pop('params')`?

Reply:
As a utility function, I choose not to mutate elements in `pgroups`.